src/type_checker/checker.rs

```diff
@@ -484,6 +487,10 @@ impl<B: Backend> TypeChecker<B> {
                 let el_typ = match typ.typ {
                     TyKind::Array(typkind, _) => *typkind,
                     TyKind::GenericSizedArray(typkind, _) => *typkind,
+                    TyKind::Tuple(typs) => match &idx.kind {
+                        ExprKind::BigUInt(index) => typs[index.to_usize().unwrap()].clone(),
```

Comment:
This should return an error `TupleIndexOutofBounds` if the index is out of bound.